core/src/main/java/org/apache/iceberg/SystemProperties.java

```diff
@@ -18,7 +18,9 @@
  */
 package org.apache.iceberg;
 
-/** Configuration properties that are controlled by Java system properties. */
+/**
+ * Configuration properties that are controlled by Java system properties or environmental variable
```

Comment:
This class is now a bit confusing, should we just make another class?  

I dont think there's that many places that need this, that we need to always call these utils?

Reply:
What about adding a new class `RuntimeInfo` to handle both properties and env? And deprecate `SystemProperties`.

Reply:
Yea that sounds cleaner to me.  Curious will there ever be a case we don't want env variables but just rely on system properties?

Reply:
I think there shouldn't be that case, we can don't set the env variable and also the property variable has higher priority.

Reply:
Using both system property and environment variable to control some settings are very common usage, for example: `ARROW_ENABLE_NULL_CHECK_FOR_GET` and `arrow.enable_null_check_for_get`.